component/prometheus/kubernetes/podmonitors/configGenerator.go

```diff
@@ -0,0 +1,176 @@
+package podmonitors
+
+// SEE https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go
+
+import (
+	"fmt"
+	"regexp"
+
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	commonConfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	promk8s "github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/model/relabel"
+)
+
+type configGenerator struct {
+	config *Arguments
+}
+
+// the k8s sd config is mostly dependent on our local config for accessing the kubernetes cluster.
+// if undefined it will default to an in-cluster config
+func (cg *configGenerator) generateK8SSDConfig(namespaceSelector v1.NamespaceSelector, namespace string, role promk8s.Role, attachMetadata *v1.AttachMetadata) *promk8s.SDConfig {
```

Comment:
I would really like to see unit tests for each of these methods instead of just having the one massive test for generatePodMonitorConfig.